### PR TITLE
feat(i18n): Allow label rewrite via advanced branding

### PR DESCRIPTION
**Pull request by vp8x8**

### CODE REVIEW

File: react/features/base/i18n/i18next.js [hunk 1]
```diff
@@ -46,6 +49,8 @@ const options = {
     load: 'languageOnly',
     ns: [ 'main', 'languages', 'countries' ],
     react: {
+        // re-render when a new resource bundle is added
+        bindI18nStore: 'added',
```
Comment (quitrk) [inline react/features/base/i18n/i18next.js:53]: I'm confused as to how is this working?

File: react/features/base/i18n/functions.js [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+/**
+ * Changes the main translation bundle.
+ *
+ * @param {string} language - The language e.g. 'en', 'fr'.
+ * @param {string} options.url - The url of the translation bundle.
+ * @param {Object} options.data - The actual bundle.
+ * @returns {void}
+ */
+export async function changeLanguageBundle(language: string, { url, data }: {url?: string, data?: Object}) {
+    let bundle = data;
```
Comment (quitrk) [inline react/features/base/i18n/functions.js:17]: what's the point of initializing with `data`?

Comment (quitrk) [reply react/features/base/i18n/functions.js:17]: seems like we never pass it anyway

File: react/features/base/i18n/actionTypes.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * The type of (redux) action which signals that language has been changed.
+ */
+export const LANGUAGE_CHANGED = 'LANGUAGE_CHANGED';
+
+/**
+ * The type of (redux) action which signals that a dynamic translation bundle will be used.
+ */
+export const USE_DYNAMIC_LANGUAGE_BUNDLES = 'USE_DYNAMIC_LANGUAGE_BUNDLES';
```
Comment (quitrk) [inline react/features/base/i18n/actionTypes.js:16]: this doesn't seem to be used

File: react/features/base/i18n/i18next.js (hunk 1 above)
Comment (vp8x8) [reply react/features/base/i18n/i18next.js:53]: All the labels/strings are stored separately in i18next's store. Whenever a new bundle(which contains labels/strings) is added an `added` event is emitted, which forces a re-render - remember we have acces to `t` using their HOC or hook.

File: react/features/base/i18n/actionTypes.js (hunk 3 above)
Comment (vp8x8) [reply react/features/base/i18n/actionTypes.js:16]: Yep, leftover. Forgot to delete.

File: react/features/base/i18n/functions.js (hunk 2 above)
Comment (vp8x8) [reply react/features/base/i18n/functions.js]: Removed it

**Review by quitrk — DISMISSED** [review]

File: react/features/base/i18n/i18next.js [hunk 4]
```diff
@@ -87,4 +92,12 @@ i18next.addResourceBundle(
 // since i18next is not yet initialized at that point.
 require('./BuiltinLanguages');
 
+i18next.on('initialized', () => {
+    APP.store.dispatch({ type: I18NEXT_INITIALIZED });
```
Comment (saghul) [inline react/features/base/i18n/i18next.js:96]: This won't work on mobile.

Comment (vp8x8) [reply react/features/base/i18n/i18next.js]: Indeed. Good catch. Fixed

**Comment by vp8x8** [comment]
Jenkins test this please

**Review by quitrk — APPROVED** [review]

**Pull request merged**